packages/react-instantsearch-dom/src/widgets/RangeSlider.js

```diff
@@ -29,14 +29,17 @@ class Range extends React.Component {
 
   state = { currentValues: { min: this.props.min, max: this.props.max } };
 
-  componentWillReceiveProps(sliderState) {
-    // @TODO: Derived State, maybe render
-    if (sliderState.canRefine) {
+  componentDidUpdate(prevProps) {
```

Comment:
I also think that this should be derived state, to make sure it happens before render

Reply:
See https://github.com/algolia/react-instantsearch/pull/2288#discussion_r273374037 for more information.